### DISCUSSION TITLE
Generator generates unwanted fromJson fields

### DISCUSSION THREAD

**Discussion by kirqe** [simolus3/drift#1853] [Q&A] [answered]
There're a few fields where I don't want to set values manually when importing from json

The generator still generates the following `id: serializer.fromJson<int>(json['id']),` for the field below. 

`IntColumn get id => integer().autoIncrement()();`

This leads to errors during importing of json records(the value is null error) How can I stop the generator from generating fromJson for the fields with the default values?

I collect companions from json and then insert them through db.dao.. I remember it was possible to do so in moor.

```dart
    deserializer.postsCompanions.forEach((companion) async {
      await db.postsDao.insertOrUpdate(companion);
    });
```

**Comment by simolus3** [comment, marked as answer]
As far as I remember, the default json serialization has always affected all fields.

The problem is that, with null-safety, it's not possible to _not_ read the id from json. It's non-nullable, and since `fromJson` returns a full row there has to be some source for a id. Being an auto-incrementing field, that source is usually the database. But in `fromJson` we can't refer to the database, and so it needs to come from json.

If you want to deeply customize JSON serialization, you can use [custom row classes](https://drift.simonbinder.eu/docs/advanced-features/custom_row_classes/) and let a builder with better support for JSON like `json_serializable` take care of it. That could possibly have a nullable id for a non-nullable column just so that JSON works.
